Cython/Compiler/Code.py

```diff
@@ -2374,6 +2414,39 @@ def intern_identifier(self, text):
     def get_cached_constants_writer(self, target=None):
         return self.globalstate.get_cached_constants_writer(target)
 
+    def name_in_module_state(self, cname):
```

Comment:
This seems to depend on the scope rather than the CodeWriter. Can these helper methods be moved to Symtab.py?

I noticed this when I saw that the type test code generation in `PyrexTypes.py` now needs the code object – just to identify the current scope level.

There's probably an issue with (string/int/…) constants then, but I wonder if constants aren't special enough to break the rules in some way…

Reply:
I've moved some of the logic to symtab.

I still think it's easier to access them through the code writer, although that's mostly just avoiding rewriting the same attribute lookups each time.